src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -315,19 +315,21 @@ public async Task SendMailAsync_CanBeCanceled_CancellationToken()
 
             // The server will introduce some fake latency so that the operation can be canceled before the request completes
             ManualResetEvent serverMre = new ManualResetEvent(false);
-            server.OnConnected += _ => serverMre.WaitOne();
-
             CancellationTokenSource cts = new CancellationTokenSource();
+            
+            server.OnConnected += _ => 
+            {
+                // Cancel the operation during actual server communication
+                cts.Cancel();
+                // Allow the server to continue after a brief delay
+                Task.Delay(100).ContinueWith(_ => serverMre.Set());
+            };
 
             var message = new MailMessage("foo@internet.com", "bar@internet.com", "Foo", "Bar");
 
-            Task sendTask = Task.Run(() => client.SendMailAsync(message, cts.Token));
-
-            cts.Cancel();
-            await Task.Delay(500);
-            serverMre.Set();
+            Task sendTask = client.SendMailAsync(message, cts.Token);
```

Comment:
Keep the original invocation intact.

```suggestion
            Task sendTask = Task.Run(() => client.SendMailAsync(message, cts.Token));
```